### PR TITLE
Update `SmokeTestDecoratorTest`

### PR DESCRIPTION
**Pull request by landonxjames**
## Motivation and Context

Add tests the were skipped in https://github.com/smithy-lang/smithy-rs/pull/3836 in favor of merging and unblocking customer. 

## Description

Testing our smoketest generator with the `UseDualStack` and `UseFips` endpoint built-ins. Previously the smoketest generator assumed that these were always available, but they are only available if those built-ins are included in the endpoints rule set of the model. We now test the smoke test with both of these built-ins, with each by itself, and with neither. 

## Checklist
Just a test change, no changelog

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SmokeTestsDecoratorTest.kt [hunk 1]
```diff
@@ -109,7 +227,7 @@ class SmokeTestsDecoratorTest {
             // setting enabled by default in `.crate/config.toml` in test workspaces.
             // To work around this issue, we unset `RUSTFLAGS` to allow unused and dead code.
```
Comment (ysaito1001) [inline aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SmokeTestsDecoratorTest.kt:228]: As discussed offline, can we add a clarifying comment something along the lines of

> to perform a compilation-only test, we don't need to set `mapOf(Pair("RUSTFLAGS", "--cfg smoketests"))`?

I think it's worth clarifying.

**Review by Velfi — APPROVED** [review]

File: aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SmokeTestsDecoratorTest.kt [hunk 2]
```diff
@@ -94,11 +189,34 @@ class SmokeTestsDecoratorTest {
 
             @error("server")
             structure FooException { }
-            """.asSmithyModel(smithyVersion = "2")
+        """
+        val model = (imports + traitsWithAllBuiltIns + serviceDef).asSmithyModel(smithyVersion = "2")
+        val modelWithNoDualStack = (imports + traitsWithNoDualStack + serviceDef).asSmithyModel(smithyVersion = "2")
+        val modelWithNoFips = (imports + traitsWithNoFips + serviceDef).asSmithyModel(smithyVersion = "2")
+        val modelWithNeitherBuiltIn = (imports + traitsWithNeither + serviceDef).asSmithyModel(smithyVersion = "2")
```
Comment (ysaito1001) [inline aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SmokeTestsDecoratorTest.kt:196]: 👍

**Review by ysaito1001 — APPROVED** [review]
Thanks for updating the test!

File: aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SmokeTestsDecoratorTest.kt (hunk 1 above)
Comment (landonxjames) [reply aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SmokeTestsDecoratorTest.kt]: Added clarifying comment in https://github.com/smithy-lang/smithy-rs/pull/3840/commits/c20b7b8d1d3f6eeb0ad03e4ccfc4420eff777413

**Pull request merged**